src/networking/SendQueueListener.java

```diff
@@ -0,0 +1,250 @@
+            return (value > 0) && (value <= 65535);
+        }
+        return false;
+      
+    }
+
+
+    /**
+     * The method will check whether ip:port address is valid or not.
+     * @param destination, is ip, port address as string
+     * @return boolean, return true if destination address is valid, false if it's not valid.
+     */
+    private boolean isValidAddress(String destination){
+        String[] dest = splitAddress(destination);
+        return ( dest.length == 2 && isValidIpaddress(dest[0]) && isValidPort(dest[1]) );
+    }
+
+
+    /**
+     * This method will do the work of taking the data from the queue 
+     * and send it over the network.
+     */
+    public void run(){
+        
+        /**
+         * when the the thread is started running we logged the instance of it.
+         */
+
+        logger.log(ModuleID.NETWORKING, LogLevel.INFO, "Send Queue Listener thread started running");
+
+        /**
+         * run the while loop as long as the application is running.
+         */
+        while(LanCommunicator.getStatus()){
+            
+            /**
+             * Check whether queue is empty or not, if it's not empty, 
+             * then try to send the message.
+             */
+
+            if( !SendQueue.isEmpty() ) {
+
+                /**
+                 * Take the outgoing packet from the sendqueue.
+                 */
+                OutgoingPacket out = SendQueue.front();
+
+                String destination = out.getDestination();
+
+                /**
+                 * if port number or ip address is invalid then no need to send the message
+                 * simply discard it by contiuing to the next message.
+                 */
+                if(! isValidAddress(destination)){
+                    /**
+                     * as the message is invalid simply dequeue and continue for next one.
+                     */
+                    SendQueue.dequeue();
+                    continue;
+                }
+
+                /**
+                 * Divide the destination address into ip and port from destination.
+                 */
+                String[] dest = this.splitAddress(destination);
+
+                /**
+                 * Take ip address from the dest array
+                 */
+                String ip = dest[0];
+
+                /**
+                 * take port number from the dest array and parse it into integer.
+                 */
+                int port = Integer.parseInt(dest[1]);
+                
+                /**
+                 * store the message into the message variable
+                 */
+                String message = out.getMessage();
+            
+                /**
+                 * store the identifier variable into the identifier.
+                 */
+                String identifier = out.getIdentifier();
+
+                message = identifier + ":" + message;
+                
```

Comment:
I feel we need to go with a better protocol. Currently, we just have an identifier and a message and it might seem okay to use this but it's not robust. It doesn't provide any flexibility and is prone to errors as well.
Why not use JSON? @sravand123 Kindly communicate with the team to make the changes. JSON support is present in Java and it will be much better if we use that.